include/libhal/pointers.hpp

```diff
@@ -1046,16 +1046,12 @@ class optional_ptr
   /**
    * @brief Default constructor creates a disengaged optional
    */
-  constexpr optional_ptr() noexcept
-  {
-  }
+  constexpr optional_ptr() noexcept = default;
 
   /**
    * @brief Constructor for nullptr (creates a disengaged optional)
    */
-  constexpr optional_ptr(std::nullptr_t) noexcept
-  {
-  }
+  constexpr optional_ptr(std::nullptr_t) noexcept = default;
```

Comment:
Whoops, I approved that a bit too early. This cannot be defaulted due to the input parameter, if I'm not mistaken.